### DISCUSSION TITLE
How to load diff .env file for diff env at runtime

### DISCUSSION THREAD

**Discussion by pawan-qa** [pytest-dev/pytest#11755] [Q&A]
I want to load, say `qa.env` file for `qa` env and `dev.env` file for `dev` env. How do I tell pytest to parse env name from cli arg and **load** specific `.env `file ? 

How do I dynamically pass `env` value to `find_dotenv` method-  `load_dotenv(find_dotenv(f'{env}.env'))`

My cli arg looks like `pytest -sv <test-folder-path> --env qa`

`conftest.py`

> @pytest.fixture(scope='session',autouse=True)
> def env(request):
>     return request.config.getoption('--env')
> 
> def pytest_addoption(parser):
>     parser.addoption("--env", action="store",default='qa',choices=("qa", "dev"))

I also tried with this fixture in conftest.py file -

> @pytest.fixture(scope='session',autouse=True)
> def get_env(env):
>     load_dotenv(find_dotenv(f'{env}.env'))

I am open to any solution which loads specific .env file dynamically.

**Comment by RonnyPfannschmidt** [comment]
Pytest itself is unaware of dotenv files

No built-in support planned

**Comment by nicoddemus** [comment]
@pawan-qa the code you shown should be working, do you get any errors? Make sure `get_env` is being called correctly, one way to do it is calling `breakpoint()` at the start, and checking the value of the `env` parameter.

**Reply by pawan-qa** [reply]
@nicoddemus  I tried with breakpoint as you have suggested but this `get_env` fixture does not get executed at all. Execution doesn't break at breakpoint. Pls help me with any alternate solution to load .env file dynamically.

Is there a way to fetch cli arg in `pytest_sessionstart` hook ?

**Reply by nicoddemus** [reply]
Is the fixture using `autouse=True`?

**Reply by pawan-qa** [reply]
@nicoddemus  Yes,  autouse = True
